spec/rubocop/cop/rails/blank_spec.rb

```diff
@@ -0,0 +1,264 @@
+# frozen_string_literal: true
+
+describe RuboCop::Cop::Rails::Blank, :config do
+  subject(:cop) { described_class.new(config) }
+  let(:cop_config) do
+    { 'NilOrEmpty' => true,
+      'NotPresent' => true,
+      'UnlessPresent' => true }
+  end
+
+  it 'accepts checking nil?' do
+    inspect_source(cop, 'foo.nil?')
+
+    expect(cop.offenses).to be_empty
+  end
+
+  it 'accepts checking empty?' do
+    inspect_source(cop, 'foo.empty?')
+
+    expect(cop.offenses).to be_empty
+  end
+
+  it 'accepts checking nil? || empty? on different objects' do
+    inspect_source(cop, 'foo.nil? || bar.empty?')
+
+    expect(cop.offenses).to be_empty
+  end
+
+  shared_examples :nil_or_empty? do |source|
+    it 'registers an offense' do
+      inspect_source(cop, source)
+
+      expect(cop.messages).to eq([described_class::MSG_NIL_OR_EMPTY])
+      expect(cop.highlights).to eq([source])
+    end
+
+    it 'auto-corrects' do
+      new_source = autocorrect_source(cop, source)
+
+      expect(new_source).to eq('foo.blank?')
+    end
+  end
+
+  it_behaves_like :nil_or_empty?, 'foo.nil? || foo.empty?'
+  it_behaves_like :nil_or_empty?, 'foo == nil || foo.empty?'
+  it_behaves_like :nil_or_empty?, 'nil == foo || foo.empty?'
+  it_behaves_like :nil_or_empty?, '!foo || foo.empty?'
+
+  it_behaves_like :nil_or_empty?, 'foo.nil? || !!foo.empty?'
+  it_behaves_like :nil_or_empty?, 'foo == nil || !!foo.empty?'
+  it_behaves_like :nil_or_empty?, 'nil == foo || !!foo.empty?'
+
+  shared_examples :all_variables do |variable|
+    let(:source) { "#{variable}.nil? || #{variable}.empty?" }
+
+    it "registers an offense when the variable is #{variable}" do
+      inspect_source(cop, source)
+
+      expect(cop.messages).to eq([described_class::MSG_NIL_OR_EMPTY])
+      expect(cop.highlights).to eq([source])
+    end
+
+    it "autocorrect to #{variable}.blank?" do
+      new_source = autocorrect_source(cop, source)
+
+      expect(new_source).to eq("#{variable}.blank?")
+    end
+  end
+
+  it_behaves_like :all_variables, 'foo'
+  it_behaves_like :all_variables, 'foo.bar'
+  it_behaves_like :all_variables, 'FOO'
+  it_behaves_like :all_variables, 'Foo'
+  it_behaves_like :all_variables, '@foo'
+  it_behaves_like :all_variables, '$foo'
+  it_behaves_like :all_variables, '@@foo'
+  it_behaves_like :all_variables, 'foo[bar]'
+  it_behaves_like :all_variables, 'Foo::Bar'
+
+  context 'present?' do
```

Comment:
Shouldn't the top-level contexts reflect the configuration options? I find the current structure of the tests a bit random and unclear.